datalad/support/annexrepo.py

```diff
@@ -3459,7 +3463,8 @@ def _proc_json_record(self, j):
             # use the action report to build a stable progress bar ID
             pbar_id = 'annexprogress-{}-{}'.format(
                 id(self),
-                hash(frozenset(j['action'])))
+                hash(frozenset((k, v) for k, v in j['action'].items()
+                               if isinstance(v, Hashable))))
```

Comment:
I wonder if it wouldn't start "breeding" progress bars whenever some unrelated field (`status` etc) would change?  should we may be explicitly consider only `key` and `file` keys?

Reply:
> I wonder if it wouldn't start "breeding" progress bars whenever some unrelated field (`status` etc) would change?  should we may be explicitly consider only `key` and `file` keys?

Based on the mention of "status", I think you're wondering about the wrong record.  Perhaps I could have worded the message of cd136bb6a4 to make things clearer.  `_proc_json_record` is limited to handling git-annex's byte-progress records:

```json
{
  "byte-progress": 10620441,
  "action": {
    "command": "get",
    "note": "from origin...",
    "input": [
      "."
    ],
    "key": "MD5E-...",
    "file": "..."
  },
  "total-size": 102374852,
  "percent-progress": "10.37%"
}
```

Can you think of an "action" field in the latter that would be problematic?  If not, I would prefer to stick with what I think was the code's original intention and keep this fix a minimal change.

Reply:
thanks for the detail!  I don't know what could become problematic, but I would not be completely surprised if e.g. `note` would somehow change or get omitted in subsequent records or a new field gets added -- I think it should not happen but is there a guarantee that it would not? And we do not have a dedicated test for this assumption so breakage would go unnoticed if `git-annex` decides to change something. Since of interest here are only `command`, `key` and `file` fields it still feels to me that it might be better to just consider those to not rely on assumptions.

Reply:
It cuts both ways.  Deciding to pick out certain keys is adding its own assumptions and requires updates if things change on git-annex's end.  Unless we have some indication that the "action" field as a whole isn't a reliable link across byte-progress records (I haven't found anything in output or code), I'd vote to leave it as is.

Reply:
Agree with @kyleam. Seems much more likely for `action` fields to change generally (which is covered by this approach) than a change that would result in `action` varying across progress records. The point is to associate a progress report with a particular action after all.

Reply:
Ok, 2 against 1, so let's proceed ;-)